homeassistant/components/pure_energie/__init__.py

```diff
@@ -7,13 +7,14 @@
 from homeassistant.core import HomeAssistant
 from homeassistant.exceptions import ConfigEntryNotReady
 
-from .const import DOMAIN
 from .coordinator import PureEnergieDataUpdateCoordinator
 
-PLATFORMS = [Platform.SENSOR]
+PLATFORMS: list[Platform] = [Platform.SENSOR]
```

Comment:
Skipped the coordinator because of this error message:

`ImportError: cannot import name 'PureEnergieConfigEntry' from partially initialized module 'homeassistant.components.pure_energie' (most likely due to a circular import)`

Reply:
That doesn't work because we don't use the coordinator in the init just for type checking, we actually call it.

Reply:
It is an inferred type. IMHO removing is fine 👍

Reply:
> That doesn't work because we don't use the coordinator in the init just for type checking, we actually call it.

Ah but then you can normally just move the configentry type to the coordinator.